common4j/src/main/com/microsoft/identity/common/java/logging/Logger.java

```diff
@@ -457,5 +446,26 @@ private static String getUTCDateTimeAsString() {
 
         return dateFormat.format(new Date());
     }
+
+    /**
+     * Get only the required metadata from the DiagnosticContext to plug it in the log lines.
+     * Here thread_name is taken from DiagnosticContext, and correlationId from input parameter.
+     * The need for this is because DiagnosticContext contains additional metadata which is not always required to be logged.
+     *
+     * @return String The concatenation of thread_name and correlation_id to serve as the required metadata in the log lines.
+     */
+    private static synchronized String getDiagnosticContextMetadata(String correlationId) {
+        String threadName = DiagnosticContext.INSTANCE.getRequestContext().get(DiagnosticContext.THREAD_NAME);
```

Comment:
nit: final

Reply:
can't make threadName final, as it can be set to "UNSET" in the code that follows.